app/code/community/Reverb/ReverbSync/Helper/Orders/Retrieval.php

```diff
@@ -35,39 +35,68 @@ public function queueReverbOrderSyncActions()
             throw new Reverb_ReverbSync_Model_Exception_Deactivated_Order_Sync($exception_message);
         }
 
-        $reverbOrdersJsonObject = $this->_retrieveOrdersJsonFromReverb();
+        return $this->_retrieveAndQueueOrders();
+    }
+
+    protected function _retrieveAndQueueOrders()
+    {
+        $api_url_path_to_call = $this->_getDefaultOrderRetrievalApiUrlPath();
 
-        if (!is_object($reverbOrdersJsonObject))
+        do
         {
-            return false;
-        }
+            $reverbOrdersJsonObject = $this->_retrieveOrdersJsonFromReverb($api_url_path_to_call);
 
-        $orders_array = $reverbOrdersJsonObject->orders;
+            if (!is_object($reverbOrdersJsonObject))
+            {
+                return false;
+            }
 
-        if (!is_array($orders_array))
-        {
-            return false;
-        }
+            $orders_array = $reverbOrdersJsonObject->orders;
 
-        foreach ($orders_array as $orderDataObject)
-        {
-            try
+            if (!is_array($orders_array))
             {
-                $this->_attemptToQueueMagentoOrderActions($orderDataObject);
+                return false;
             }
-            catch(Exception $e)
+
+            foreach ($orders_array as $orderDataObject)
             {
-                $order_sync_action = $this->getOrderSyncAction();
-                $error_message = $this->__(self::EXCEPTION_QUEUE_MAGENTO_ORDER_ACTION, $order_sync_action, $e->getMessage(), json_encode($orderDataObject));
-                $this->_logError($error_message);
-                $exceptionToLog = new Exception($error_message);
-                Mage::logException($exceptionToLog);
+                try
+                {
+                    $this->_attemptToQueueMagentoOrderActions($orderDataObject);
+                }
+                catch(Exception $e)
+                {
+                    $order_sync_action = $this->getOrderSyncAction();
+                    $error_message = $this->__(self::EXCEPTION_QUEUE_MAGENTO_ORDER_ACTION, $order_sync_action, $e->getMessage(), json_encode($orderDataObject));
+                    $this->_logError($error_message);
+                    $exceptionToLog = new Exception($error_message);
+                    Mage::logException($exceptionToLog);
+                }
             }
+
+            $api_url_path_to_call = $this->_getNextPageOfResultsUrlPath($reverbOrdersJsonObject);
         }
+        while(!empty($api_url_path_to_call));
 
         return true;
     }
 
+    /**
+     * Returns the url path for the next page of orders results if more results exist
+     *
+     * @param stdClass $reverbOrdersJsonObject
+     * @return string
+     */
+    protected function _getNextPageOfResultsUrlPath($reverbOrdersJsonObject)
+    {
+        if(isset($reverbOrdersJsonObject->_links->next->href))
```

Comment:
is this a safe object traversal? if links->next is nil, will it not blow up when you try to reference href?